### DISCUSSION TITLE
kB and KiB, shouldn't we standardize the units?

### DISCUSSION THREAD

**Discussion by kanarus** [rust-lang/crates.io#7177] [General]
In crates.io, `kB` is used to specify the size of package :

![Screenshot from 2023-09-26 03-17-01](https://github.com/rust-lang/cargo/assets/100482447/3d68155c-cdf6-42d6-b301-36391d6788df)

But `cargo publish` uses `KiB` :

![Screenshot from 2023-09-26 03-24-59](https://github.com/rust-lang/cargo/assets/100482447/d9776426-8688-4d20-81b3-f1656e7d986f)

This inconsistency may be strange for crate developers. Shouldn't we standardize them?

---

Then, in my opinion, changing crates.io to use `KiB` is better than the opposite. Because `git push`, almost all crate developers will do before `cargo publish`, uses `KiB/s`, not `kB/s` :

![Screenshot from 2023-09-26 04-00-46](https://github.com/rust-lang/crates.io/assets/100482447/75d8c00e-4e0c-4278-928d-9580e4e0c43a)

So I think it's a good idea in terms of consistency for crate developers. What are your thoughts on this?

**Comment by Andrew15-5** [comment]
My humble rant:

Yes! I strongly believe in binary prefixes because everything is coded in binary! I hate this inconsistency between SI and binary prefixes across not only programs, but OSes too! It's great when Megabytes and Mebibytes are indeed show different number of bytes, but it's all for nothing when user thinks that Megabytes are basically Mebibytes (every teacher/professor these days still uses SI prefixes even in the context of binary prefixes).

How many people are actually counting 8 bits times 1000 times 1000? Maybe only if you want to sell a 2TB HDD (where 2TB maybe are indeed 2 TB, but definitely not 2 TiB). And then everyone gets frustrated that they got about 1.82 TiB (`numfmt 2T --from=auto --to=iec-i --format=%.2f`).

The configuration seems very complicated as I don't use JS (only TS), never used ember etc. But the easiest solution is to change this line:

https://github.com/rust-lang/crates.io/blob/b96ed5c6ef8a54d44e5edaf53bd8827323d89c9b/app/helpers/pretty-bytes.js#L5

to:

```js
export default helper(([bytes], options) => prettyBytes(bytes, {...options, binary: true}));
```

The owner of the `pretty-bytes` package (which is how the bytes are converted on crates.io) "feels" the other way.

https://github.com/sindresorhus/pretty-bytes/issues/7#issuecomment-66918867:

> No. It's intentionally option-less. I feel base-10 is the right way to display sizes for humans.
> 
> You're of course free to use any of the other file size prettifiers if you disagree :)

And after that, the package somehow got an _option_ to use binary prefixes instead. Which means that crates.io doesn't have to look for "other file size prettifiers".

TL;DR: to counter the legacy of confusing SI prefixes in binary world, we must switch digital size measuring (for files, archives, "crates") to binary prefixes everywhere. Thankfully, a bunch of FOSS software already uses them and that is great.

**Reply by kanarus** [reply]
Thank you for your response and detailed research! Yes, that looks correct. I'll go ahead and create a pull request...

**Reply by kanarus** [reply]
Done!
- https://github.com/rust-lang/crates.io/pull/7218

waiting a review...
